### PR TITLE
Deployment documentation refresh

### PR DESCRIPTION
**Pull request by jkrogers18**

### CODE REVIEW

File: deployment/README.md [hunk 1]
```diff
@@ -1,379 +1,466 @@
...
+           sudo apt-add-repository "deb [arch=amd64] https://apt.releases.hashicorp.com $(lsb_release -cs) main" && \
+           sudo apt-get update && sudo apt-get install terraform=0.12.29
+         ```
+    - Install [Go 1.14+](https://golang.org/doc/install), for example:
+         ```shell
+         sudo apt install wget -y && \
+           wget https://golang.org/dl/go1.15.6.linux-amd64.tar.gz && \
+           sudo tar -C /usr/local -xzf go1.15.6.linux-amd64.tar.gz && \
+           export PATH=$PATH:/usr/local/go/bin
+         ```
```
Comment (zohrehj) [inline deployment/README.md:136]: why is this needed? I went through these steps on my mac and don't recall installing go.

File: deployment/scripts/set_env_var.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+#This script sets environment variables that are used throughout the deployment
```
Comment (zohrehj) [inline deployment/scripts/set_env_var.sh:2]: nit: single space after `#`

also missing copyright header.

File: deployment/README.md [hunk 3]
```diff
@@ -1,379 +1,466 @@
...
-    Make sure in your first deployment in a new folder, `enable_gcs_backend` in
-    `$MYSTUDIES_TEMPLATE` is set to `false` or commented out.
-
+1. Authenticate as a user with the permissions described above (this deployment assumes gcloud and Terraform commands are made as a user, rather than a service account)
+    - Login and update your [application default credentials](https://cloud.google.com/docs/authentication/production), for example you could run `gcloud auth login --update-adc` (when using a Google Compute Engine VM you must update the application default credentials, otherwise requests will continue to be made with its default service account)
+    - Remember to run `gcloud auth revoke` to log your user account out once your deployment is complete
+
+### Set up your CICD pipelines
+
+1. Generate your Terraform configuration files
```
Comment (zohrehj) [inline deployment/README.md:159]: this should be only done after the devops project has been created and a copy of the terraform state exists locally; I don't think it is the case here.

File: deployment/README.md [hunk 4]
```diff
@@ -1,379 +1,466 @@
...
+    - Set the `enable_gcs_backend` flag in [`mystudies.hcl`](/deployment/mystudies.hcl) to `false`, for example:
+         ```bash
```
Comment (zohrehj) [inline deployment/README.md:161]: I think changing the values manually is probably easier for most people. Using commands is more error prone and may not behave the same way in all environments/OS.

File: deployment/README.md [hunk 5]
```diff
@@ -1,379 +1,466 @@
...
+
+    > Note: If your pre-submit checks or post-submit `terraform apply` fail with an error related to billing accounts, you may not have the [quota](https://support.google.com/cloud/answer/6330231?hl=en) necessary to attach all of your projects to the specified billing account. You may need to [request additional quota](https://support.google.com/code/contact/billing_quota_increase).
+1. [Grant](https://cloud.google.com/iam/docs/granting-changing-revoking-access) the [`roles/owner`](https://cloud.google.com/resource-manager/docs/access-control-proj#using_basic_roles) permission to the `{PREFIX}-{ENV}-project-owners@{DOMAIN}` group for each of your newly created projects
+
+### Update your domain’s DNS records
+
+1. [View created DNS zones](https://console.cloud.google.com/net-services/dns/zones) in your `{PREFIX}-{ENV}-apps` project
```
Comment (zohrehj) [inline deployment/README.md:208]: thanks for adding these

File: deployment/README.md [hunk 6]
```diff
@@ -1,379 +1,466 @@
...
+         tfengine --config_path=$ENGINE_CONFIG --output_path=$GIT_ROOT/deployment/terraform
+         git checkout -b sql-bucket-permissions
+         git add $GIT_ROOT/deployment/terraform
+         git commit -m "Set SQL bucket permissions"
+         git push origin sql-bucket-permissions
+         ```
+    - Once your pull request pre-submit checks have completed successfully, and you have received code review approval, merge your pull request to trigger `terraform apply`(this may take up to 10 minutes - you can view the status of the operation in the [Cloud Build history](https://console.cloud.google.com/cloud-build/builds) of your `devops` project)
+1. Initialize your MySQL databases by importing SQL scripts
+    - Upload the necessary SQL script files to the `{PREFIX}-{ENV}-mystudies-sql-import` storage bucket that you created during Terraform deployment, for example:
+         ```bash
```
Comment (zohrehj) [inline deployment/README.md:245]: the git commands above assume that the user is within the git directory, these ones assume that the user is outside of the directory. I think it makes more sense to explicitly ask the user to navigate to their local git directory and then assume all following commands are run from that location.

File: deployment/README.md [hunk 7]
```diff
@@ -1,379 +1,466 @@
...
+    `manual-mystudies-from-email-address` | The return email address that is shown is system-generated messages (for example, no-reply@example.com) | Set this value now or enter a placeholder | `echo -n "<SECRET_VALUE>" \| gcloud secrets versions add "manual-mystudies-from-email-address" --data-file=-`
+    `manual-mystudies-from-email-domain` | The domain of the above email address (just the value after “@”) | Set this value now or enter a placeholder | `echo -n "<SECRET_VALUE>" \| gcloud secrets versions add "manual-mystudies-from-email-domain" --data-file=-`
+    `manual-mystudies-smtp-hostname` | The hostname for your email account’s SMTP server (for example, smtp.gmail.com) | Set this value now or enter a placeholder | `echo -n "<SECRET_VALUE>" \| gcloud secrets versions add "manual-mystudies-smtp-hostname" --data-file=-`
+    `manual-mystudies-smtp-use-ip-allowlist` | Typically ‘False’; if ‘True’, the platform will not authenticate to the email server and will rely on the allowlist configured in the SMTP service | Set this value now or enter a placeholder | `echo -n "<SECRET_VALUE>" \| gcloud secrets versions add "manual-mystudies-smtp-use-ip-allowlist" --data-file=-`
```
Comment (zohrehj) [inline deployment/README.md:331]: nit: lowercase false and true

File: deployment/README.md [hunk 8]
```diff
@@ -1,379 +1,466 @@
...
+    `manual-android-deeplink-url` | The URL to redirect to after Android login (for example, `app://{PREFIX}.{DOMAIN}/mystudies`) | Set now if you know what this value will be - otherwise create a placeholder and update after completing your [Android](/Android/) deployment (leave as placeholder if you will be deploying to iOS only) | `echo -n "<SECRET_VALUE>" \| gcloud secrets versions add "manual-android-deeplink-url" --data-file=-`
+    `manual-ios-bundle-id` | The value you will obtain from Xcode (Project target > General tab > Identity section > Bundle identifier) during [iOS configuration](/iOS/) | Set now if you know what this value will be - otherwise create a placeholder and update after completing your [iOS](/iOS/) deployment (leave as placeholder if you will be deploying to Android only) | `echo -n "<SECRET_VALUE>" \| gcloud secrets versions add "manual-ios-bundle-id" --data-file=-`
```
Comment (zohrehj) [inline deployment/README.md:341]: For a prod application bundle ID needs to be verified with Apple, it's usually a reverse domain name that you own; it is a unique app identifier and application capabilities are mapped to this value.
for more details see https://developer.apple.com/documentation/appstoreconnectapi/bundle_ids

File: deployment/README.md [hunk 9]
```diff
@@ -1,379 +1,466 @@
...
+         kubectl apply \
+           -f $GIT_ROOT/deployment/kubernetes/cert.yaml \
+           -f $GIT_ROOT/deployment/kubernetes/ingress.yaml
+         ```
+    - Update firewalls:
+        - Run `kubectl describe ingress $PREFIX-$ENV`
+        - Look at the suggested commands under "Events", in the form of "Firewall
+        change required by network admin"
+        - Run each of the suggested commands
+1. Check the [Kubernetes dashboard](https://console.cloud.google.com/kubernetes/workload) in your `{PREFIX}-{ENV}-apps` project to view the status of your deployment
```
Comment (zohrehj) [inline deployment/README.md:403]: it would be a good idea to ask users to wait until all apps are healthy, since the rest of the steps require healthy K8s pods and application to successfully run.

File: deployment/README.md [hunk 10]
```diff
@@ -1,379 +1,466 @@
...
     ```
+<!--
+### Troubleshooting
 
-### Step 13: Clean up
-
-1. Revoke your super admin access by running `gcloud auth revoke` and
-    authenticate as a normal user for daily activities.
+See the [*Troubleshooting*](/documentation/troubleshooting.md) guide for more information.
+--->
```
Comment (zohrehj) [inline deployment/README.md:464]: let's just remove this part instead of leaving it commented out

**Review by zohrehj — COMMENTED** [review]
This is a huge improvement, thanks for adding this.
I added a few comments, please take a look and address if needed.

**Comment by zohrehj** [comment]
One more comment, across this PR, you are using <prefix>.<domain>, it should be <prefix>-<env>.<domain>, see [this line](https://github.com/GoogleCloudPlatform/fda-mystudies/blob/master/deployment/mystudies.hcl#L631)

File: deployment/README.md (hunk 1 above)
Comment (jkrogers18) [reply deployment/README.md:136]: I think this was in the original instructions that Xin put together. If it's not necessary I will remove.

File: deployment/scripts/set_env_var.sh (hunk 2 above)
Comment (jkrogers18) [reply deployment/scripts/set_env_var.sh]: fixed

File: deployment/README.md (hunk 10 above)
Comment (jkrogers18) [reply deployment/README.md:464]: will do

File: deployment/README.md (hunk 9 above)
Comment (jkrogers18) [reply deployment/README.md:403]: adding

File: deployment/README.md (hunk 8 above)
Comment (jkrogers18) [reply deployment/README.md:341]: added

File: deployment/README.md (hunk 7 above)
Comment (jkrogers18) [reply deployment/README.md:331]: fixed

File: deployment/README.md (hunk 6 above)
Comment (jkrogers18) [reply deployment/README.md:245]: added cd $GIT_ROOT to the relevant code blocks

File: deployment/README.md (hunk 3 above)
Comment (jkrogers18) [reply deployment/README.md:159]: updated the title to "Create your devops project and configure CICD pipelines" since I believe this section explains how to do both
